lib/experimental/full-client-side-navigation.php

```diff
@@ -0,0 +1,31 @@
+<?php
+/**
+ * Registers full client-side navigation option using the Interactivity API and adds the necessary directives.
+ */
+
+// Add the full client-side navigation config option.
+wp_interactivity_config( 'core/router', array( 'fullClientSideNavigation' => true ) );
+
+// Add directives to all links.
+// This should probably be done per site, not by default when this option is enabled.
+function gutenberg_add_client_side_navigation_directives( $content, $block ) {
+	// Don't add directives to query blocks and pagination blocks.
+	if (
+		'core/query' === $block['blockName'] ||
+		'core/query-pagination-next' === $block['blockName'] ||
+		'core/query-pagination-previous' === $block['blockName'] ||
+		'core/query-pagination-numbers' === $block['blockName']
+		) {
+		return $content;
+	}
+	$p = new WP_HTML_Tag_Processor( $content );
+	while ( $p->next_tag( array( 'tag_name' => 'a' ) ) ) {
```

Comment:
What is the expectation for the external URLs? What about links pointing to an anchor on the same page?

Reply:
Those cases are currently handled by the navigate function:
* It checks if the URL is external [here](https://github.com/WordPress/gutenberg/pull/59707/files#diff-c6f39cecb38b1631b9cfa08afbfefd80805da6e69fc2b8eea4868c911dc601e1R121-R126).
* I just added some logic to scroll to the anchor if exists [here](https://github.com/WordPress/gutenberg/pull/59707/files#diff-c6f39cecb38b1631b9cfa08afbfefd80805da6e69fc2b8eea4868c911dc601e1R158-R163).

Reply:
Nice, thank yiy for confirming 👍🏻 

One additional consideration, should that behavior get only applied to blocks marked as interactive through `supports.interactivity`?

Reply:
Mmm not sure if for the experiment we should apply it only to interactive blocks. For example, we might want to add full page client-side navigation to images or links included in the footer, which are not interactive.